Sources/Danger/Plugins/SwiftLint/SwiftLint.swift

```diff
@@ -46,86 +46,115 @@ extension SwiftLint {
+                                                directory: directory,
+                                                arguments: &arguments,
+                                                shellExecutor: shellExecutor,
+                                                swiftlintPath: swiftlintPath,
+                                                failAction: failAction)
+        }
 
-            let outputJSON = shellExecutor.execute(swiftlintPath,
-                                                   arguments: arguments,
-                                                   environmentVariables: inputFiles)
-            violations = makeViolations(from: outputJSON, failAction: failAction)
+        guard !violations.isEmpty else {
+            return []
         }
 
         let currentPath = currentPathProvider.currentPath
         violations = violations.map { violation in
-            let updatedPath = violation.file.deletingPrefix(currentPath).deletingPrefix("/")
             var violation = violation
+
+            let updatedPath = violation.file.deletingPrefix(currentPath).deletingPrefix("/")
             violation.file = updatedPath
+
             if strict {
                 violation.severity = .error
             }
             return violation
         }
 
-        if !violations.isEmpty {
-            if inline {
-                violations.forEach { violation in
-                    switch violation.severity {
-                    case .error:
-                        failInlineAction(violation.messageText, violation.file, violation.line)
-                    case .warning:
-                        warnInlineAction(violation.messageText, violation.file, violation.line)
-                    }
+        if inline {
+            violations.forEach { violation in
+                switch violation.severity {
+                case .error:
+                    failInlineAction(violation.messageText, violation.file, violation.line)
+                case .warning:
+                    warnInlineAction(violation.messageText, violation.file, violation.line)
                 }
-            } else {
-                var markdownMessage = """
-                ### SwiftLint found issues
-
-                | Severity | File | Reason |
-                | -------- | ---- | ------ |\n
-                """
-                markdownMessage += violations.map { $0.toMarkdown() }.joined(separator: "\n")
-                markdownAction(markdownMessage)
             }
+        } else {
+            var markdownMessage = """
+            ### SwiftLint found issues
+
+            | Severity | File | Reason |
+            | -------- | ---- | ------ |\n
+            """
+            markdownMessage += violations.map { $0.toMarkdown() }.joined(separator: "\n")
+            markdownAction(markdownMessage)
         }
 
         return violations
     }
 
+    private static func lintAll(directory: String?,
+                                arguments: inout [String],
```

Comment:
Why do you need arguments to be `inout`?
You can make `arguments` a var inside the function, given you don't need the updated value back

Reply:
You are absolutely right. Thank you for spotting this Xcode "extract method" relic. :)